src/Maths/Silk.NET.Maths/Vector2.cs

```diff
@@ -268,145 +258,340 @@ public static void BaryCentric
+            {
+                return (Vector128<T>)(object)Vector128.Create((double)(object)X, (double)(object)Y);
+            }
+
+            Scalar<T>.ThrowInvalidType();
+            return default;
+        }
+
+        public Vector256<T> AsVector256()
+        {
+            if (typeof(T) == typeof(byte))
+            {
+                return (Vector256<T>)(object)Vector256.Create((byte)(object)X, (byte)(object)Y, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0);
+            }
+
+            if (typeof(T) == typeof(sbyte))
+            {
+                return (Vector256<T>)(object)Vector256.Create((sbyte)(object)X, (sbyte)(object)Y, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0);
+            }
+
+            if (typeof(T) == typeof(ushort))
+            {
+                return (Vector256<T>)(object)Vector256.Create((ushort)(object)X, (ushort)(object)Y, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0);
+            }
+
+            if (typeof(T) == typeof(short))
+            {
+                return (Vector256<T>)(object)Vector256.Create((short)(object)X, (short)(object)Y, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0);
+            }
+
+            if (typeof(T) == typeof(uint))
+            {
+                return (Vector256<T>)(object)Vector256.Create((uint)(object)X, (uint)(object)Y, 0, 0, 0, 0, 0, 0);
+            }
+
+            if (typeof(T) == typeof(int))
+            {
+                return (Vector256<T>)(object)Vector256.Create((int)(object)X, (int)(object)Y, 0, 0, 0, 0, 0, 0);
+            }
+
+            if (typeof(T) == typeof(ulong))
+            {
+                return (Vector256<T>)(object)Vector256.Create((ulong)(object)X, (ulong)(object)Y, 0, 0);
+            }
+
+            if (typeof(T) == typeof(long))
+            {
+                return (Vector256<T>)(object)Vector256.Create((long)(object)X, (long)(object)Y, 0, 0);
+            }
+
+            /*if (typeof(T) == typeof(Half))
+            {
+                retrn (Vector256<T>)(object)Vector256.Create((Half)(object)X, (Half)(object)Y, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0);
+            }*/
+
+            if (typeof(T) == typeof(float))
+            {
+                return (Vector256<T>)(object)Vector256.Create((float)(object)X, (float)(object)Y, 0, 0, 0, 0, 0, 0);
+            }
+
+            if (typeof(T) == typeof(double))
+            {
+                return (Vector256<T>)(object)Vector256.Create((double)(object)X, (double)(object)Y, 0, 0);
+            }
+
+            Scalar<T>.ThrowInvalidType();
+            return default;
         }
 #endif
 
 #if BTEC_INTRINSICS
-        public System.Numerics.Vector<T> AsVector()
+        public unsafe Vector<T> AsVector()
         {
-            throw new NotImplementedException();
+            Vector<T> vec = new Vector<T>();
+            Unsafe.Copy(&vec, ref X);
```

Comment:
Vector\<T\> does not have a size guarantee and may not be able to fit the vector, so be sure to check on Vector\<T\>.Count before filling it.

Also, I recommend using the official constructors as I presume the various HWIntrinsics won't be used during load if filled like this (could be wrong though, better safe than sorry however)

Reply:
I thought Vector<T> is guaranteed 128 bits? And as Vector<T> is 128 bits at mac itself, this seemed reasonable. Will check tomorrow

Reply:
![image](https://user-images.githubusercontent.com/11160611/84700701-937be600-af4b-11ea-80bd-3207ed672df5.png)